app/gui/view/graph-editor/src/component/node/input/widget/label.rs

```diff
@@ -90,9 +92,10 @@ impl SpanWidget for Widget {
             parent_port_hovered <- widgets_frp.hovered_port_children.map(move |h| h.contains(&id));
             parent_port_hovered <- parent_port_hovered.on_change();
             text_color <- frp.text_color.on_change();
-            label_color <- all_with3(
-                &style, &text_color, &parent_port_hovered,
-                |style, state, hovered| state.to_color(*hovered, style)
+            text_alpha <- frp.text_alpha.on_change();
+            label_color <- all_with4(
+                &style, &text_color, &parent_port_hovered, &text_alpha,
+                |style, state, hovered, alpha_factor| state.to_color(*hovered, style, alpha_factor.to_alpha(style))
```

Comment:
This is very minor, but given that we are already using custom `AlphaState` type here, I would rather pass it directly to `to_color`, then do the `to_alpha` step inside. The `style` is already passed in anyway.

I would also consider just using a simple `pending: bool` here to remove some code, but I guess the enum makes it easy to add more complicated alpha-related logic later.

```suggestion
                |style, state, hovered, alpha_factor| state.to_color(*hovered, style, alpha_factor)
```